src/gossip/fuzz_service.zig

```diff
@@ -272,73 +251,148 @@ pub fn run(args: *std.process.ArgIterator) !void {
+            );
+
+            // this is mainly used to just send packets through the fuzzer
+            // but we also want to respond to pings so we need to run the full gossip service
+            const fuzz_handle = try std.Thread.spawn(.{}, GossipService.run, .{
+                &gossip_service_fuzzer, .{
+                    .spy_node = true,
+                    .dump = false,
+                },
+            });
+            fuzz_handle.detach();
+
+            break :blk .{ gossip_service_client, gossip_service_client.packet_incoming_channel, client_handle };
+        } else {
+            var gossip_service_fuzzer = try GossipService.init(
+                allocator,
+                allocator,
+                fuzz_contact_info,
+                fuzz_keypair,
+                (&SocketAddr.fromEndpoint(&to_entrypoint))[0..1], // we only want to communicate with one node
+                &exit,
+                .noop, // no logs
+            );
+
+            // this is mainly used to just send packets through the fuzzer
+            // but we also want to respond to pings so we need to run the full gossip service
+            const fuzz_handle = try std.Thread.spawn(.{}, GossipService.run, .{
+                &gossip_service_fuzzer, .{
+                    .spy_node = true,
+                    .dump = false,
+                },
+            });
+
+            break :blk .{ gossip_service_fuzzer, gossip_service_fuzzer.packet_outgoing_channel, fuzz_handle };
+        }
+    };
 
-    const fuzz_handle = try std.Thread.spawn(.{}, GossipService.run, .{
-        &gossip_service_fuzzer, .{
-            .spy_node = true,
-            .dump = false,
-        },
-    });
+    // NOTE: this is useful when we want to run for an inf amount of time and want to
+    // early exit at some point without killing the process
+    var fuzzing_loop_exit = AtomicBool.init(false);
+    // wait for any keyboard input to exit early
+    var exit_handle = try std.Thread.spawn(.{}, waitForExit, .{&fuzzing_loop_exit});
+    exit_handle.detach();
 
-    const SLEEP_TIME = 0;
-    // const SLEEP_TIME = std.time.ns_per_ms * 10;
-    // const SLEEP_TIME = std.time.ns_per_s;
+    // start fuzzing
+    try fuzz(
+        allocator,
+        &fuzzing_loop_exit,
+        maybe_max_messages,
+        &rng,
+        &fuzz_keypair,
+        LegacyContactInfo.fromContactInfo(&fuzz_contact_info),
+        to_entrypoint,
+        packet_channel,
+    );
 
-    // wait for keyboard input to exit
-    var loop_exit = AtomicBool.init(false);
-    var exit_handle = try std.Thread.spawn(.{}, waitForExit, .{&loop_exit});
+    // cleanup
+    std.debug.print("\t=> shutting down...\n", .{});
+    exit.store(true, .unordered);
+    handle.join();
+    gossip_client.deinit();
+    std.debug.print("\t=> done.\n", .{});
+}
 
+pub fn fuzz(
+    allocator: std.mem.Allocator,
+    loop_exit: *AtomicBool,
+    maybe_max_messages: ?usize,
+    rng: *std.rand.DefaultPrng,
```

Comment:
Is there a reason this isn't `std.Random`? If it's taking advantage of the specific properties of `Xoshiro256`, it should take that, instead of the `DefaultPrng` alias which could conceivably be changed to a different implementation.

Reply:
1ddc1eb